src/c_api/c_api.cc

```diff
@@ -972,6 +972,44 @@ int MXNDArrayCreateSparseEx(int storage_type,
       Context::Create(static_cast<Context::DeviceType>(dev_type), dev_id),
       delay_alloc != 0,
       dtype, aux_types, aux_shapes);
+}
+
+int MXNDArrayCreateSparseEx(int storage_type,
+                            const uint32_t *shape,
+                            uint32_t ndim,
+                            int dev_type,
+                            int dev_id,
+                            int delay_alloc,
+                            int dtype,
+                            uint32_t num_aux,
+                            int *aux_type,
+                            uint32_t *aux_ndims,
+                            const uint32_t *aux_shape,
+                            NDArrayHandle *out) {
+  API_BEGIN();
+  CreateSparseNDArray<uint32_t, uint32_t>(storage_type, shape, ndim, dev_type, dev_id,
```

Comment:
Can we also change the ndim type here from uint32_t to int? It looks strange to me why 32-bit version uses uint32_t while 64-bit version uses int.